src/DependencyInjection/CompilerPass/AutoRegisterChainMiddlewareCompilerPass.php

```diff
@@ -73,9 +73,12 @@ private function discoverModels(ContainerBuilder $container): array
         $messagesHandler = $container->findTaggedServiceIds($this->handlerTagName);
 
         $models = [];
-        foreach ($messagesHandler as $tags) {
+        foreach ($messagesHandler as $serviceId => $tags) {
+            $handlerClass = $container->getDefinition($serviceId)->getClass();
+            $messageClass = substr($handlerClass, 0, -7);
+
             foreach ($tags as $tag) {
```

Comment:
Probably the same thing that for `AutoRegisterMessageHandlerCompilerPass` but the fact that a handler is consider as being a middleware, in addition of the "bindings" and "models" notions that are unclear, I'm not entirely sure.

Reply:
Les middleware en vont pas évoluer dans cette mr. je vais créer un fichier de documentation des middleware séparer dans la documentation.

Reply:
please resolve